### PR TITLE
Update to access API for tokens

### PR DESCRIPTION
**Pull request by nickshine**
* Updates dependencies
* Switches token creation from the older /security/tokens API to the access/v1/tokens API, which is necessary for tokens scoped to other services like X-Ray.

### CODE REVIEW

**Review by ahuang-splunk — DISMISSED** [review]

**Review by mbos-splunk — DISMISSED** [review]

**Review by dan1 — DISMISSED** [review]
LGTM

File: plugin/artifactory_client.go [hunk 1]
```diff
@@ -60,37 +65,60 @@ func NewClient(config *ConfigStorageEntry) (Client, error) {
...
+
 	if err != nil {
-		return nil, fmt.Errorf("failed to build artifactory service config - %v", err.Error())
+		return nil, fmt.Errorf("failed to build artifactory service config - %w", err)
 	}
 
 	client, err := artifactory.New(artifactoryServiceConfig)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to build artifactory client- %v", err.Error())
+		return nil, fmt.Errorf("Failed to build artifactory client- %w", err)
```
Comment (dstopka) [inline plugin/artifactory_client.go:101]:
```suggestion
		return nil, fmt.Errorf("failed to build artifactory client - %w", err)
```

File: plugin/artifactory_client.go [hunk 2]
```diff
@@ -60,37 +65,60 @@ func NewClient(config *ConfigStorageEntry) (Client, error) {
...
 	ac.client = client
 
+	accessServicesConfig, err := artconfig.NewConfigBuilder().
+		SetServiceDetails(accessDetails).
+		SetContext(context.Background()).
+		SetThreads(1).
+		Build()
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to build access service config - %w", err)
```
Comment (dstopka) [inline plugin/artifactory_client.go:113]:
```suggestion
		return nil, fmt.Errorf("failed to build access service config - %w", err)
```

File: plugin/artifactory_client.go [hunk 3]
```diff
@@ -60,37 +65,60 @@ func NewClient(config *ConfigStorageEntry) (Client, error) {
...
+	}
+
+	accessClient, err := access.New(accessServicesConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to build access client - %w", err)
```
Comment (dstopka) [inline plugin/artifactory_client.go:118]:
```suggestion
		return nil, fmt.Errorf("failed to build access client - %w", err)
```

File: plugin/path_config.go [hunk 4]
```diff
@@ -85,18 +81,13 @@ func (backend *ArtifactoryBackend) pathConfigWrite(ctx context.Context, req *log
 	}
 
 	if baseURL, ok := data.GetOk("base_url"); ok {
-		url := appendTrailingSlash(baseURL.(string))
-		cfg.BaseURL = url
+		cfg.BaseURL = baseURL.(string)
```
Comment (dstopka) [inline plugin/path_config.go:84]: Do you know what was the reason for `appendTrailingSlash` being used here?

Comment (nickshine) [reply plugin/path_config.go:84]: Artifactory will fail to start up if the given URL does not have a trailing slash, but it is handled in a different function now.

**Review by dstopka — APPROVED** [review]

**Pull request merged**